phoenix-core/src/main/java/org/apache/phoenix/mapreduce/AbstractBulkLoadTool.java

```diff
@@ -244,14 +254,26 @@ private int loadData(Configuration conf, CommandLine cmdLine) throws Exception {
         PTable table = PhoenixRuntime.getTable(conn, qualifiedTableName);
         tablesToBeLoaded.add(new TargetTableRef(qualifiedTableName, table.getPhysicalName().getString()));
         boolean hasLocalIndexes = false;
+        boolean hasGlobalIndexes = false;
         for(PTable index: table.getIndexes()) {
             if (index.getIndexType() == IndexType.LOCAL) {
                 hasLocalIndexes =
                         qualifiedIndexTableName == null ? true : index.getTableName().getString()
                                 .equals(qualifiedIndexTableName);
-                if (hasLocalIndexes) break;
+                if (hasLocalIndexes && hasGlobalIndexes) break;
+            }
+            if (index.getIndexType() == IndexType.GLOBAL) {
+                hasGlobalIndexes = true;
+                if (hasLocalIndexes && hasGlobalIndexes) break;
             }
         }
+
+        if(hasGlobalIndexes && tableNotEmpty && !cmdLine.hasOption(ENABLE_CORRUPT_INDEXES.getOpt())){
+            throw new IllegalStateException("Bulk Loader error: Bulk loading is disabled for" +
```

Comment:
A bit more context may help:

Bulk Loading error: Bulk loading is disabled for non empty tables with global indexes, because it will corrupt the global index table in most cases.
Use the --corruptindexes option to override this check.